### PR TITLE
chore: misc lints

### PR DESCRIPTION
**Pull request by WillLillis**

### CODE REVIEW

File: lib/binding_rust/lib.rs [hunk 1]
```diff
@@ -692,15 +692,15 @@ impl Parser {
...
+            let (callback, text) = payload.cast::<(&mut F, Option<T>)>().as_mut().unwrap();
             *text = Some(callback(byte_offset as usize, position.into()));
             let slice = text.as_ref().unwrap().as_ref();
             *bytes_read = slice.len() as u32;
             slice.as_ptr().cast::<c_char>()
         }
 
         let c_input = ffi::TSInput {
-            payload: &mut payload as *mut (&mut F, Option<T>) as *mut c_void,
+            payload: std::ptr::addr_of_mut!(payload).cast::<c_void>(),
```
Comment (ObserverOfTime) [inline lib/binding_rust/lib.rs:703]:
```suggestion
            payload: core::ptr::addr_of_mut!(payload).cast::<c_void>(),
```

File: lib/binding_rust/lib.rs [hunk 2]
```diff
@@ -794,7 +794,7 @@ impl Parser {
...
```
Comment (ObserverOfTime) [inline lib/binding_rust/lib.rs:797]:
```suggestion
            payload: core::ptr::addr_of_mut!(payload).cast::<c_void>(),
```

File: lib/binding_rust/lib.rs [hunk 3]
```diff
@@ -865,10 +865,10 @@ impl Parser {
             ));
             let slice = text.as_ref().unwrap().as_ref();
             *bytes_read = slice.len() as u32 * 2;
-            slice.as_ptr() as *const c_char
+            slice.as_ptr().cast::<c_char>()
         }
         let c_input = ffi::TSInput {
-            payload: &mut payload as *mut (&mut F, Option<T>) as *mut c_void,
+            payload: std::ptr::addr_of_mut!(payload).cast::<c_void>(),
```
Comment (ObserverOfTime) [inline lib/binding_rust/lib.rs:871]:
```suggestion
            payload: core::ptr::addr_of_mut!(payload).cast::<c_void>(),
```

**Review by ObserverOfTime — CHANGES_REQUESTED** [review]

File: cli/src/main.rs [hunk 4]
```diff
@@ -974,6 +979,7 @@ impl Test {
         }
 
         // For the rest of the queries, find their tests and run them
+        #[allow(clippy::redundant_closure_for_method_calls)]
```
Comment (amaanq) [inline cli/src/main.rs:982]:
```suggestion
```

File: highlight/src/lib.rs [hunk 5]
```diff
@@ -234,6 +234,7 @@ impl<'query, 'tree: 'query, T: TextProvider<I>, I: AsRef<[u8]>> Iterator
                     m.as_mut_ptr(),
                     core::ptr::addr_of_mut!(capture_index),
                 ) {
+                    #[allow(clippy::transmute_undefined_repr)]
```
Comment (amaanq) [inline highlight/src/lib.rs:237]:
```suggestion
```

File: highlight/src/lib.rs [hunk 6]
```diff
@@ -576,6 +577,7 @@ impl<'a> HighlightIterLayer<'a> {
                     mem::transmute::<&mut QueryCursor, &'static mut QueryCursor>(&mut cursor)
                 };
                 let captures = unsafe {
+                    #[allow(clippy::transmute_undefined_repr)]
```
Comment (amaanq) [inline highlight/src/lib.rs:580]:
```suggestion
```

File: xtask/src/main.rs [hunk 7]
```diff
@@ -21,14 +21,11 @@ fn main() -> Result<(), Box<dyn std::error::Error>> {
...
-        }
-        _ => {
-            println!("invalid task: {task}");
-            std::process::exit(1);
-        }
+    if task.as_str() == "bump-version" {
+        bump_versions()?;
+    } else {
+        println!("invalid task: {task}");
+        std::process::exit(1);
```
Comment (amaanq) [inline xtask/src/main.rs:28]: leave this alone for now - being reworked

File: cli/src/fuzz/mod.rs [hunk 8]
```diff
@@ -1,3 +1,4 @@
+#![allow(clippy::ref_option)]
```
Comment (amaanq) [inline cli/src/fuzz/mod.rs:1]: we don't want to allow this

```suggestion
```

File: cli/loader/src/lib.rs [hunk 9]
```diff
@@ -366,6 +368,7 @@ impl<'a> CompileConfig<'a> {
     }
 }
 
+#[allow(clippy::non_send_fields_in_send_ty)]
```
Comment (amaanq) [inline cli/loader/src/lib.rs:371]: tbh, `Send` needs to be removed for loader

File: cli/generate/src/render.rs [hunk 10]
```diff
@@ -735,6 +735,7 @@ impl Generator {
         let mut leading_simple_transition_count = 0;
         let mut leading_simple_transition_range_count = 0;
         for (chars, action) in &state.advance_actions {
+            #[allow(clippy::checked_conversions)]
```
Comment (amaanq) [inline cli/generate/src/render.rs:738]:
```suggestion
```

File: cli/generate/src/node_types.rs [hunk 11]
```diff
@@ -499,6 +499,7 @@ pub fn generate_node_types_json(
         } else if !syntax_grammar.variables_to_inline.contains(&symbol) {
             // If a rule is aliased under multiple names, then its information
             // contributes to multiple entries in the final JSON.
+            #[allow(clippy::or_fun_call)]
```
Comment (amaanq) [inline cli/generate/src/node_types.rs:502]: Upstream should fix the lint popping up here

```suggestion
```

File: cli/benches/benchmark.rs [hunk 12]
```diff
@@ -1,3 +1,4 @@
+#![allow(clippy::ref_option)]
```
Comment (amaanq) [inline cli/benches/benchmark.rs:1]:
```suggestion
```

File: cli/loader/src/lib.rs (hunk 9 above)
Comment (amaanq) [reply cli/loader/src/lib.rs]: remove

**Pull request merged**